cpp/src/arrow/buffer_test.cc

```diff
@@ -1014,4 +1014,12 @@ TYPED_TEST(TypedTestBuffer, ResizeOOM) {
 #endif
 }
 
+TEST(TestBufferConcatenation, EmptyBuffer) {
+  const std::string contents = "hello, world";
```

Comment:
Add a comment explaining the test motivation here
```suggestion
  // GH-36913: UB shouldn't be triggered by copying from a null pointer
  const std::string contents = "hello, world";
```